File: apis/elysium-character-api/src/main/java/com/seventh_root/elysium/api/character/ElysiumCharacter.java
package com.seventh_root.elysium.api.character;

import com.seventh_root.elysium.api.player.ElysiumPlayer;
import com.seventh_root.elysium.core.database.TableRow;

public interface ElysiumCharacter extends TableRow {

    ElysiumPlayer getPlayer();

    void setPlayer(ElysiumPlayer player);

    String getName();

    void setName(String name);

    Gender getGender();

    void setGender(Gender gender);

    int getAge();

    void setAge(int age);

    Race getRace();

    void setRace(Race race);

    String getDescription();

    void setDescription(String description);

    boolean isDead();

    void setDead(boolean dead);

    double getHealth();

    void setHealth(double health);

    double getMaxHealth();

    void setMaxHealth(double maxHealth);

    int getMana();

    void setMana(int mana);

    int getMaxMana();

    void setMaxMana(int maxMana);

    int getFoodLevel();

    void setFoodLevel(int foodLevel);

    int getThirstLevel();

    void setThirstLevel(int thirstLevel);

}

File: apis/elysium-character-api/src/main/kotlin/com/seventh_root/elysium/api/character/ElysiumCharacter.kt
package com.seventh_root.elysium.api.character

import com.seventh_root.elysium.api.player.ElysiumPlayer
import com.seventh_root.elysium.core.database.TableRow

interface ElysiumCharacter : TableRow {

    var player: ElysiumPlayer?
    var name: String
    var gender: Gender?
    var age: Int
    var race: Race?
    var description: String
    var isDead: Boolean
    var health: Double
    var maxHealth: Double
    var mana: Int
    var maxMana: Int
    var foodLevel: Int
    var thirstLevel: Int

}
